examples/plot_nested-cv.py

```diff
@@ -0,0 +1,151 @@
+"""
+=============================================================================
+Non-nested versus nested cross-validation for estimating prediction intervals
+=============================================================================
+
+This example compares non-nested and nested cross-validation strategies for
+estimating prediction intervals with :class:`mapie.estimators.MapieRegressor`.
+
+In the regular sequential method, a cross-validation parameter search is
+carried out over the entire training set.
+The model with the set of parameter that gives the best score is then used in
+MAPIE to estimate the prediction intervals associated with the predictions.
+A limitation of this method is that residuals used by MAPIE are computed on
+validation dataset, which can be subject to overfitting as far as hyperparameter
+tuning is concerned.
+This fools MAPIE in being slightly too optimistic with confidence intervals.
+
+To solve this problem, an alternative option is to perform a nested
+cross-validation parameter search directly within the MAPIE estimator on each
+*out-of-fold* dataset.
+For each testing fold used by MAPIE to store residuals, an internal cross-validation
+occurs on the training fold, optimizing hyperparameters.
+This ensures that residuals seen by MAPIE are never seen by the algorithm beforehand.
+However, this method is much computationally heavier since it results in
+:math:`N \times P` calculations, where *N* is the number of *out-of-fold*
+models and *P* the number of parameter search iterations, versus :math:`N + P`
+for the non-nested approach.
+
+Here, we compare the two strategies on the Boston dataset. We use the Random
+Forest Regressor as a base regressor for the CV+ method. For the sake of light
+computation, we adopt a RandomizedSearchCV parameter search strategy with a low
+number of iterations and with a reproducible random state.
+
+It is found that estimating prediction intervals via the CV+ method with the nested
+and non-nested approaches gives very similar scores and identical effective coverages.
+In practice, the nested cross-validated parameter search approach gives *out-of-fold*
+models which have slightly different best parameters.
+The two approaches therefore give slightly different predictions with the nested CV approach
+estimating slightly larger prediction interval widths by a few percents at most (apart from a
+handful of exceptions).
+For this exemple, the two approaches result in identical scores and in identical effective
+coverages.
```

Comment:
Il manque une phrase de conclusion claire du style :
```
In the general case, the recommended approach is to use nested cross-validation, since it does not understimate residuals and hence prediction intervals. However, in this particular example, effective coverages of both nested and non-nested methods are the same.

Reply:
Phrase ajoutée.